src/Core/src/Hosting/IMauiHandlersServiceProvider.cs

```diff
@@ -1,14 +1,17 @@
 #nullable enable
 using System;
+using Microsoft.Maui.Hosting;
 
 namespace Microsoft.Maui
 {
-	public interface IMauiHandlersServiceProvider : IServiceProvider
+	public interface IMauiHandlersServiceProvider : IMauiServiceProvider
 	{
 		Type? GetHandlerType(Type iview);
 
 		IViewHandler? GetHandler(Type type);
 
 		IViewHandler? GetHandler<T>() where T : IView;
+
+		IMauiHandlersCollection GetCollection();
```

Comment:
I decided to embrace the dynamic nature of handlers, since we are also going to maybe use this, and made it a real feature.

Reply:
I changed my mind. We are not that dynamic. This is on a separate interface for now until we decide if this is a good thing: https://github.com/dotnet/maui/pull/851